phy/cluster/manual/clustering.py

```diff
@@ -6,15 +6,14 @@
 # Imports
 #------------------------------------------------------------------------------
 
-from collections import namedtuple, defaultdict, OrderedDict
-from copy import deepcopy
-
 import numpy as np
 
-from ...ext.six import iterkeys, itervalues, iteritems
 from ...utils.array import _as_array, _is_array_like
```

Comment:
It is indeed clearer when reading the code, but it sometimes leads to hard-to-fix runtime problems like circular imports, etc. You can find various points of views on the Internet. It seems that historically absolute imports were highly recommended by PEPs and Guido, but that's no longer the case.